src/lib/io/include/sol-uart.h

```diff
@@ -53,36 +53,47 @@ extern "C" {
 
 struct sol_uart;
 
-struct sol_uart *sol_uart_open(const char *port_name);
+enum sol_uart_speed {
+    SOL_UART_SPEED_9600 = 0,
+    SOL_UART_SPEED_19200,
+    SOL_UART_SPEED_38400,
+    SOL_UART_SPEED_57600,
+    SOL_UART_SPEED_115200
+};
+
+enum sol_uart_data_bits {
+    SOL_UART_DATA_BITS_8 = 0,
+    SOL_UART_DATA_BITS_7,
+    SOL_UART_DATA_BITS_6,
+    SOL_UART_DATA_BITS_5
+};
+
+enum sol_uart_parity {
+  SOL_UART_PARITY_DISABLE = 0,
+  SOL_UART_PARITY_ENABLE_EVEN,
+  SOL_UART_PARITY_ENABLE_ODD
+};
+
+enum sol_uart_stop_bits {
```

Comment:
imho follow the same pattern used in pwm and gpio. Leave the 3 (4 in the case) enums as is and have a struct called config where those are set and passed as arg. So we have a consistent API across lib/io